# 문제 1. 배열의 합 구하기

**문제 설명:**

정수로 구성된 배열이 주어졌을 때, 배열의 합을 구하는 최적의 함수를 작성하시오.

입력 예시: [1, 2, 3, 4, 5]
출력 예시: 15

In [ ]:
#length 의 절반만큼만 반복하고 양옆에서 중앙으로 오면서 각 원소의 합 을 구한다.

userinput = list(map(int, input().split()))
length = len(userinput)
sum = 0

# length가 짝수일 때
if length % 2 == 0:
    for i in range(length//2):
        sum += userinput[i] + userinput[length-i-1]
elif length % 2 == 1:
    # length가 홀수일 때
    for i in range(length//2):
        sum += userinput[i] + userinput[length-i-1]
    sum += userinput[length//2]

print(sum)

15


# 문제 2. 배열 뒤집기

**문제 설명:**

주어진 배열을 뒤집은 새로운 배열을 반환하는 함수를 2가지 이상의 구현방식으로 작성하고 이들간의 특성을 비교해보하시오.

입력 예시: [1, 2, 3, 4]
출력 예시: [4, 3, 2, 1]

In [ ]:
userinput2 = list(map(int, input().split()))
print(userinput2[::-1]) 

for i in range(len(userinput2)//2):
    userinput2[i], userinput2[-i-1] = userinput2[-i-1], userinput2[i]
print(userinput2) 

[4, 3, 2, 1]
[4, 3, 2, 1]


# 문제 3. 10진수를 2진수로 변환하기

In [9]:
userinput3 = int(input())
print(bin(userinput3)[2:]) 

1101


**문제 설명:**

0 이상의 10진수 정수를 입력 받아 2진수 문자열로 반환하는 함수를 작성하시오.

입력 예시: 13
출력 예시: '1101'

# 문제 4. 2진수를 10진수로 변환하기

**문제 설명:**

2진수 문자열을 입력 받아 10진수 정수로 반환하는 함수를 작성하시오.

입력 예시: '1101'
출력 예시: 13

In [8]:
userinput4 = input()
print(int(userinput4, 2))

13


# 문제 5. 소수 판별하기

**문제 설명:**

주어진 정수가 소수(prime number)인지 확인하는 함수를 작성하시오.

입력 예시: 17
출력 예시: True

In [ ]:
userinput5 = int(input())
is_prime = True

if userinput5 < 2: # 0과 1은 소수가 아님
    is_prime = False
else:
    for i in range(2, userinput5):
        if userinput5 % i == 0:
            is_prime = False
            break

print(is_prime)


True


# 문제 6. 주어진 범위에서 소수 찾기

**문제 설명:**

주어진 범위 [start, end]에서 모든 소수를 리스트로 반환하시오.

입력 예시: start = 10, end = 20
출력 예시: [11, 13, 17, 19]

In [ ]:
start = int(input())
end = int(input())
list_prime = []

for i in range(start, end+1):
    is_prime = True
    if i < 2:
        is_prime = False
    else:
        for j in range(2, int(i**0.5) + 1):
            if i % j == 0:
                is_prime = False
                break
    if is_prime:
        list_prime.append(i)

print(list_prime)



[2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 101, 103, 107, 109, 113, 127, 131, 137, 139, 149, 151, 157, 163, 167, 173, 179, 181, 191, 193, 197, 199, 211, 223, 227, 229, 233, 239, 241, 251, 257, 263, 269, 271, 277, 281, 283, 293, 307, 311, 313, 317, 331, 337, 347, 349, 353, 359, 367, 373, 379, 383, 389, 397, 401, 409, 419, 421, 431, 433, 439, 443, 449, 457, 461, 463, 467, 479, 487, 491, 499, 503, 509, 521, 523, 541, 547, 557, 563, 569, 571, 577, 587, 593, 599, 601, 607, 613, 617, 619, 631, 641, 643, 647, 653, 659, 661, 673, 677, 683, 691, 701, 709, 719, 727, 733, 739, 743, 751, 757, 761, 769, 773, 787, 797, 809, 811, 821, 823, 827, 829, 839, 853, 857, 859, 863, 877, 881, 883, 887, 907, 911, 919, 929, 937, 941, 947, 953, 967, 971, 977, 983, 991, 997]
